src/components/ProfilePage.tsx

```diff
@@ -105,19 +116,16 @@ export const ProfilePage = (props: ProfilePageProps) => {
         setEditing(false)
     }
 
-    const uploadHandler = async(e: React.ChangeEvent<HTMLInputElement>) => {
-        if(e.target.files){
-            const file = e.target.files[0]
-            try {
-                const resp: FileUploadComplete = await uploadFile(accessToken, file.name, file)
-                const picUrl = await getFileHandleByIdURL(resp.fileHandleId,accessToken)
-                setFileHandleId(resp.fileHandleId)
-                setProfilePicUrl(picUrl)
-            } catch(err:any){
-                console.error(err)
-            }
-        }
-    }
+	const onSelectFile = (e: React.ChangeEvent<HTMLInputElement>) => {
+		if (e.target.files && e.target.files.length > 0) {
+			const reader = new FileReader();
+			reader.readAsDataURL(e.target.files[0]);
+			reader.addEventListener("load", () => {
+				setImage(reader.result as string);
+                setOpenCropModal(true)
+			});
+		}
```

Comment:
I have a very cursory understanding of the FileReader API, but I think you need to attach the even listener before calling `readAsDataURL`. I suspect you have a race condition otherwise.